src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -7235,6 +7235,25 @@ class C { }
                 );
         }
 
+        [Fact, WorkItem(40136, "https://github.com/dotnet/roslyn/issues/40136")]
+        public void SelfReferencingAttribute()
```

Comment:
>SelfReferencingAttribute [](start = 20, length = 24)

Consider also testing scenarios when the attribute is applied to return type and parameters. Also when an attribute is applied to a member of the attribute (Field/Property/setter) initialized by the application.